### ISSUE TITLE
Pages not found after upgrade to 2.1.3

### ISSUE THREAD

**Issue by walderich** [bolt/bolt#3265]
After upgrading Bolt from 2.1.0 to 2.1.3 all entries with slugs containing underscores could not be accessed anymore. I now found out that "editing" the slug (i.e. click on "edit slug" button and saving back without any change) converts all underscores to minus signs. Is this the intended behaviour? Obviously the page names searched are also replaced, that's why they wheren't found in the first place. But I couldn't find any hint in the changelogs (but I didn't search very hard, too). Shouldn't the slugs be renamed after the update, or is the manual intervention intended?

**Comment by bobdenotter** [comment]
We _did_ start using a new lib to create slugs recently, but i don't think any significant changes were made in that regard between 2.1.0 and 2.1.3.. Sure you did't upgrade from an older version?

**Comment by GawainLynch** [comment]
Also there was a regression introduced somewhere late in 2.0-dev that used the underscores instead of hyphens… You can however now specify in your [config.yml](https://github.com/bolt/bolt/blob/master/app/config/config.yml.dist#L160) what replacement character you want to use.

That said, that shouldn't prevent you from routing to records *with* an existing underscore in the slug.

**Comment by walderich** [comment]
@bobdenotter You're right of course. My last update was on February 8th. So it must have been version 2.0.6
@GawainLynch When I saw the problem, I was logged into Bolt so I got Symphony's "Whoops" error page. When I read you correctly, you say, that visitors should have got the correct page, right?

**Comment by GawainLynch** [comment]
Oh, sorry if I communicated badly with my last sentence! I was mostly thinking out aloud that it sounds weird as existing records should **not** break and yes they should still work in the front end… the fact that it doesn't makes me scratch my head :grin:

**Comment by walderich** [comment]
What would be the safest/easiest way to fix this problem? Replace all occurences in the database with a query? Or will this be addressed by Bolt in one of the next releases?
As I looked over the current config.yml(.dist) I realized that my current configuration is still old (from the installation I guess). Am I supposed to merge it manually after upgrading? Are there any other files which need such adjustment?
